homeassistant/components/azure_data_explorer/client.py

```diff
@@ -0,0 +1,87 @@
+"""Setting up the ingest client."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import io
+import logging
+from typing import Any
+
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder
+from azure.kusto.data.data_format import DataFormat
+from azure.kusto.ingest import (
+    IngestionProperties,
+    ManagedStreamingIngestClient,
+    QueuedIngestClient,
+    StreamDescriptor,
+)
+
+from .const import (
+    CONF_ADX_CLUSTER_INGEST_URI,
+    CONF_ADX_DATABASE_NAME,
+    CONF_ADX_TABLE_NAME,
+    CONF_APP_REG_ID,
+    CONF_APP_REG_SECRET,
+    CONF_AUTHORITY_ID,
+    CONF_USE_FREE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AzureDataExplorerClient:
+    """Class for Azure Data Explorer Client."""
+
+    def __init__(self, data: Mapping[str, Any]) -> None:
+        """Create the right class."""
+
+        self._cluster_ingest_uri = data[CONF_ADX_CLUSTER_INGEST_URI]
+        self._database = data[CONF_ADX_DATABASE_NAME]
+        self._table = data[CONF_ADX_TABLE_NAME]
+        self._client_id = data[CONF_APP_REG_ID]
+        self._client_secret = data[CONF_APP_REG_SECRET]
+        self._authority_id = data[CONF_AUTHORITY_ID]
+        self._use_queued_ingestion = data[CONF_USE_FREE]
+
+        self._cluster_query_uri = self._cluster_ingest_uri.replace(
```

Comment:
Why are you doing this? IF we really require it, we should move it to the config flow, where we do it only once and not on every creation of the client

Reply:
This is to forsve the corrent non "ingest-" Url. I have removed it